lucene/src/main/java/org/hibernate/hql/lucene/internal/LuceneQueryRendererDelegate.java

```diff
@@ -67,8 +67,8 @@ protected void addSortField(PropertyPath propertyPath, String collateName, boole
 
 		SortField.Type sortType = SortField.Type.STRING;
 		FieldBridge fieldBridge = propertyHelper.getFieldBridge( targetTypeName, propertyPath.getNodeNamesWithoutAlias() );
-		if ( fieldBridge instanceof NullEncodingTwoWayFieldBridge ) {
-			fieldBridge = ( (NullEncodingTwoWayFieldBridge) fieldBridge ).unwrap();
+		if ( fieldBridge instanceof BridgeAdaptor ) {
+			fieldBridge = ( (BridgeAdaptor) fieldBridge ).unwrap( NumericFieldBridge.class );
```

Comment:
This is a bit dangerous, because if the field bridge is an adaptor over something else than a `NumericFieldBridge`, the field bridge will be null from there on. It'll work, but it's not obvious why.

Anyway... Inspecting bridges is a bit tricky. Since you seem to have access to metadata here, why don't you simply use `org.hibernate.search.engine.metadata.impl.DocumentFieldMetadata.getNumericEncodingType()` to retrieve the numeric encoding? It's also accessible through the public metadata API, mind you.